### ISSUE TITLE
Refactor Pipeline

### ISSUE THREAD

**Issue by CatStark** [jina-ai/finetuner#170]
Refactor Pipeline as suggested in Tadej [slides](https://docs.google.com/presentation/d/10w7ep475CDEd0ScPM_bCR7hM7SCbXapzxc0QWmIyJ38/edit#slide=id.p)

below *new* refers to dataset without tuples/triplets, *old* refers to user manually creates tuples/triplets

- [x] refactor `Dataset`, support new and old style dataset. - @tadejsv 
- [x] add sampler, take N items as batch from `Dataset`. - @tadejsv 
- [x] adjust get embedding (use model extract embeddings from batch of data given by sampler). - @tadejsv 
- [x] add miner, create pairs/triplets given batch of embeddings and labels. @bwanglzu 
- [x] adjust loss functions. - @bwanglzu 
- [x] transformer based text pre-processing support

we start from pytorch, then go to paddle and keras, finish by making abstraction classes.

**Comment by Dfurb87** [comment]
- [ ]
